vision6.module

```diff
@@ -608,36 +600,26 @@ function vision6_workbench_moderation_transition($node, $previous_state, $new_st
 function _vision6_node_beforesave($node) {
   global $base_url;
 
-  $wrapper = entity_metadata_wrapper('node', $node);
+  try {
+    $wrapper = entity_metadata_wrapper('node', $node);
+  }
+  catch (Exception $e) {
+    watchdog_exception('vision6', $e);
+    drupal_exit();
+  }
 
   if (!variable_get('vision6_' . $node->type, FALSE) || !user_access('access vision6 publish')) {
     return;
   }
 
-  if (!user_access('publish revisions')) {
-    if (!is_null($node->nid) && !empty($node->original)) {
-      $node->current_status = $node->original->status;
-      $node->current_title = $node->original->title;
-      $node->current_comment = $node->original->comment;
-      $node->current_promote = $node->original->promote;
-      $node->current_sticky = $node->original->sticky;
-      $node->revision_moderation = 1;
-    }
+  if (module_exists('workbench_moderation') && ($node->workbench_moderation) && $node->workbench_moderation['current']->state != 'published') {
     return;
   }
-
-  if (isset($node->workbench_moderation)) {
-    if ($node->workbench_moderation['current']->state != 'published') {
-      return;
-    }
-  }
-  else {
-    if (!$wrapper->status->value()) {
-      return;
-    }
+  elseif (!$wrapper->status->value()) {
+    return;
   }
 
-  if (isset($wrapper->field_vis6_published_as_synd)) {
+  if (!empty($node->field_vis6_published_as_synd)) {
```

Comment:
Use of $wrapper when the property of an object is not available will cause the PHP Fatal error. Using $node object instead.